src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs

```diff
@@ -422,45 +422,44 @@ private async Task<OpenApiOperation> GenerateOpenApiOperationFromMetadataAsync(A
                 }
             }
 
-            var requestContentTypes = operation.RequestBody?.Content?.Values;
+            var requestContentTypes = operation.RequestBody?.Content?.Keys;
             if (requestContentTypes is not null)
             {
-                foreach (var content in requestContentTypes)
+                foreach (var contentType in requestContentTypes)
                 {
-                    content.Encoding = new Dictionary<string, OpenApiEncoding>();
-                    var requestParameters = apiDescription.ParameterDescriptions.Where(desc => desc.IsFromBody() || desc.IsFromForm());
-                    var countOfParameters = requestParameters.Count();
-                    if (countOfParameters > 0)
+                    var contentTypeValue = operation.RequestBody.Content[contentType];
+                    var fromFormParameters = apiDescription.ParameterDescriptions.Where(desc => desc.IsFromForm());
+                    var fromFormParametersCount = fromFormParameters.Count();
```

Comment:
Looks like we end up enumerating the collection multiple times in the case where `>0`, so we should just `ToList()` or `ToArray()` after the `Where()` and then use `Length`/`Count` and remove this count variable.